core/libs/commonwealth/commonwealth/mavlink_comm/MavlinkComm.py

```diff
@@ -51,28 +68,58 @@ async def get_mavlink_message(
                 async with session.get(request_url, timeout=request_timeout) as response:
                     if not response.status == 200:
                         raise MavlinkMessageReceiveFail(f"Received status code of {response.status}.")
+                    # if message is "None", try re-detecting systemid
+                    if await response.text() == "None":
```

Comment:
I'm worried about this. Correct me if I'm wrong, but looking at the core it appears that the user does not have control over the `system id` that it gona be used with this detection logic.

If that's the case, this can result in problems, where the user restart BlueOS with multiple vehicle on the network, or it going to connect in a vehicle without knowing wich one, since it can change every time. The user should enforce the `system id`.

Reply:
The user has no control over it, but it is not enough to have multiple vehicles in the network, someone will need to actually create a misconfigured mavlink endpoint pointing to this blueos instance instead of the topside.

Reply:
I'm leaning to this approach as it handles the user simply changing SYSID_THISMAV graciously.

Reply:
Ok, we can continue as it is so.